packages/components/src/components/PipelineRun/PipelineRun.js

```diff
@@ -111,9 +111,21 @@ export /* istanbul ignore next */ class PipelineRunContainer extends Component {
 
     return taskRuns
       .map(taskRun => {
-        const taskName = taskRun.spec.taskRef.name;
-        const task = selectedTask(taskName, tasks);
-        if (!task) {
+        let taskSpec;
+
+        if (taskRun.spec.taskRef) {
+          const task = selectedTask(taskRun.spec.taskRef.name, tasks);
+
+          if (!task) {
+            return null;
+          }
+
+          taskSpec = task.spec;
+        } else {
+          ({ taskSpec } = taskRun.spec.taskSpec);
```

Comment:
```suggestion
          ({ taskSpec } = taskRun.spec);
```

Reply:
@AlanGreene thanks for pointing this out, I will take the opportunity to investigate the issue as i don't know much about conditions yet.

Reply:
@AlanGreene about the suggested change, are you sure ? the purpose here is to get a task spec, if i accept the suggested change, it will be an object containing the task spec, not the task spec itself.

Reply:
`({ taskSpec } = taskRun.spec)` is using object destructuring, so it's equivalent to
`taskSpec = taskRun.spec.taskSpec`

Reply:
👍